docs/src/examples/redis-clustering-aware.md

```diff
@@ -0,0 +1,140 @@
+# Redis Clustering with cluster aware client
+
+The following guide shows you how to configure Shotover to support proxying Redis Cluser *aware* clients to [Redis cluster](https://redis.io/topics/cluster-spec).
+
+## Overview
+
+In this example, we will be connecting to a Redis cluster that has the following topology:
+
+* `0.0.0.0:2220`
+* `0.0.0.0:2221`
+* `0.0.0.0:2222`
+* `0.0.0.0:2223`
+* `0.0.0.0:2224`
+* `0.0.0.0:2225`
+
+Shotover will listen on the loopback adapter and act as a sidecar for our application that speaks Redis. In this example we will use `redis-benchmark` with cluster mode enabled as our Redis cluster aware client application. Use the following `docker-compose.yml` to run the Redis cluster.
```

Comment:
Can we link to it and provide a copy pasteable snippet instead of embedding the whole thing here.
IMO it makes sense to embed the Topology.yaml and not embed the docker-compose.yml because:
* the docker-compose.yml is a lot longer and disrupts the flow of the article
* the Topology.yaml is what we are really trying to teach here so we want to reduce friction of reading it as much as possible.
* the docker-compose.yml is just one of multiple ways of setting up the cluster.

At the very least we should be consistent between example guides.
Even writing mostly the exact same paragraph between articles is useful as it can be easy to recognize as being the same information. And the duplication is fine because we want people to be able to start at any article that suits their use case.

Reply:
Sounds good to me. I've also realised that using the port rewrite only makes sense if you are using Shotover as a sidecar to each Redis node so I will have to redo the docker-compose.